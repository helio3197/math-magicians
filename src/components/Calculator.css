.calc-wrapper {
  width: 450px;
  padding: 12px;
  display: flex;
  flex-direction: column;
  border: 1px solid;
  gap: 8px;
  background-color: #e3e3ec;
}

.calc-result {
  padding: 16px 8px;
  border: 1px solid;
  text-align: end;
  font-size: 2rem;
  background-color: #acafba;
}

.calc-buttons {
  display: flex;
  flex-wrap: wrap;
  width: 100%;
}

.calc-buttons > button {
  width: 25%;
  padding: 16px;
  font-size: 1.5rem;
}

.calc-buttons > button:nth-last-of-type(3) {
  width: 50%;
}

.calc-buttons > button:nth-of-type(4n),
.calc-buttons > button:last-child {
  background-color: lightgray;
  border-width: 0.5px;
}

.calc-buttons > button:nth-of-type(4n):hover,
.calc-buttons > button:last-child:hover {
  background-color: #c6c6c6;
}

.calc-buttons > button:nth-of-type(4n):active,
.calc-buttons > button:last-child:active {
  background-color: #e1e1e1;
}
