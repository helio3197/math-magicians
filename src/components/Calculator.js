import { useState } from 'react';
import calculate from '../logic/calculate';
import dragElement from './drag-calc';
import './Calculator.css';

const Calculator = () => {
  const [calcObj, setCalcObj] = useState({ total: '0' });

  const buttonHandler = (e) => {
    const btnValue = e.target.innerHTML;
    setCalcObj(calculate(calcObj, btnValue));
  };

  const buttons = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];

  const { total } = calcObj;

  return (
    <div
      style={{
        width: '100%',
        height: '100%',
        padding: '24px 0',
        display: 'flex',
        flexWrap: 'wrap',
        maxWidth: '960px',
      }}
    >
      <div
        style={{
          width: '50%',
        }}
      >
        <h2>Let&apos;s do some math!</h2>
        <h3>The calculator is draggable!</h3>
      </div>
      <div
        style={{
          width: '50%',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <div onMouseDown={dragElement} role="button" tabIndex="0" className="calc-wrapper">
          <div className="calc-result">
            <output>{total}</output>
          </div>
          <div className="calc-buttons">
            {buttons.map((el) => (<button type="button" onClick={buttonHandler} key={el}>{el}</button>))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
