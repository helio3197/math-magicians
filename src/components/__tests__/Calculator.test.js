import { render, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Test for Calculator component', () => {
  test('The component should render', () => {
    const { asFragment } = render(<Calculator />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('Clicking the numbers 5, 4 and 2 the output should display 542', () => {
    const { getByTestId } = render(<Calculator />);
    fireEvent.click(getByTestId('5'));
    fireEvent.click(getByTestId('4'));
    fireEvent.click(getByTestId('2'));

    expect(getByTestId('result')).toHaveTextContent('542');
  });
});
