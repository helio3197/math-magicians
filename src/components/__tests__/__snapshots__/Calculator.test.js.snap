// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Calculator component The component should render 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; height: 100%; padding: 24px 0px; display: flex; flex-wrap: wrap; max-width: 960px;"
  >
    <div
      style="width: 50%;"
    >
      <h2>
        Let's do some math!
      </h2>
      <h3>
        The calculator is draggable!
      </h3>
    </div>
    <div
      style="width: 50%; display: flex; justify-content: center; align-items: center;"
    >
      <div
        class="calc-wrapper"
        role="button"
        tabindex="0"
      >
        <div
          class="calc-result"
        >
          <output
            data-testid="result"
          >
            0
          </output>
        </div>
        <div
          class="calc-buttons"
        >
          <button
            data-testid="AC"
            type="button"
          >
            AC
          </button>
          <button
            data-testid="+/-"
            type="button"
          >
            +/-
          </button>
          <button
            data-testid="%"
            type="button"
          >
            %
          </button>
          <button
            data-testid="รท"
            type="button"
          >
            รท
          </button>
          <button
            data-testid="7"
            type="button"
          >
            7
          </button>
          <button
            data-testid="8"
            type="button"
          >
            8
          </button>
          <button
            data-testid="9"
            type="button"
          >
            9
          </button>
          <button
            data-testid="x"
            type="button"
          >
            x
          </button>
          <button
            data-testid="4"
            type="button"
          >
            4
          </button>
          <button
            data-testid="5"
            type="button"
          >
            5
          </button>
          <button
            data-testid="6"
            type="button"
          >
            6
          </button>
          <button
            data-testid="-"
            type="button"
          >
            -
          </button>
          <button
            data-testid="1"
            type="button"
          >
            1
          </button>
          <button
            data-testid="2"
            type="button"
          >
            2
          </button>
          <button
            data-testid="3"
            type="button"
          >
            3
          </button>
          <button
            data-testid="+"
            type="button"
          >
            +
          </button>
          <button
            data-testid="0"
            type="button"
          >
            0
          </button>
          <button
            data-testid="."
            type="button"
          >
            .
          </button>
          <button
            data-testid="="
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
